modules/post/multi/gather/enum_software_versions.rb

```diff
@@ -0,0 +1,139 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Multiplatform Installed Software Version',
+        'Description' => %q{
+          This module, when run against a compromised machine, will gather details on all installed software,
+          including their versions and if available, when they were installed, and will save it into a loot file for later use.
+          Users can then use this loot file to determine what additional vulnerabilites may affect the target machine.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [ 'gwillcox-r7' ],
+        'Platform' => %w[win linux osx bsd solaris android],
+        # 'Platform'      => %w{ android osx win linux bsd solaris },
+        'SessionTypes' => [ 'meterpreter', 'shell' ]
+      )
+    )
+  end
+
+  def store_linux_loot(listing)
+    file = store_loot('host.linux.software.versions', 'text/plain', session, listing, 'installed_software.txt', 'Installed Software and Versions')
+    print_good("Stored information about the installed products to the loot file at #{file}")
+  end
+
+  # Run Method for when run command is issued
+  def run
+    case session.platform
+    when 'windows'
+      listing = cmd_exec('wmic product get Name, Description, Version, InstallDate', nil, 6000)
+      unless listing =~ /Description/
+        print_error('Was unable to get a listing of installed products...')
+        return nil
+      end
+      file = store_loot('host.windows.software.versions', 'text/plain', session, listing, 'installed_software.txt', 'Installed Software and Versions')
+      print_good("Stored information about the installed products to the loot file at #{file}")
+    when 'linux'
+      # All of the following options were taken from https://distrowatch.com/dwres.php?resource=package-management
+      operating_system = cmd_exec('hostnamectl')
+      if operating_system =~ /not found/
```

Comment:
Maybe another `case` statement?

Reply:
Hmm good point, I'll update this code tomorrow to fix this up, this should really be a `case` statement as you mentioned 👍

Reply:
You should ask @gwillcox-r7 for permission to change his code!

Reply:
Hmm so the one problem I do see with this is this:

```
2.6.6 :001 > string = "debian"
2.6.6 :002 > case string
2.6.6 :003 > when /ubuntu/i || /debian/i
2.6.6 :004 >   print "A"
2.6.6 :005 > end
 => nil 
2.6.6 :006 > case string
2.6.6 :007 > when /debian/i
2.6.6 :008 >   print "A"
2.6.6 :009 > end
A => nil 
2.6.6 :010 > case string
2.6.6 :011 > when /ubuntu/i || /debian/i
2.6.6 :012 >   print "A"
2.6.6 :013 > end
 => nil 
2.6.6 :014 > 
```
I might possibly be missing something here but it looks like one can't do multiple regex in one line with a `when` statement.

Reply:
You can use a comma!

Reply:
Thanks @wvu-r7, works like a charm, much appreciated! Will implement this now 👍